actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java

```diff
@@ -231,6 +309,264 @@ private void addUserCourseBatch(Request actorMessage) {
+          || (!batchCreatorRootOrgId.equals(participantWithRootOrgIds.get(userId)))) {
+        throw new ProjectCommonException(
+            ResponseCode.invalidCourseCreatorId.getErrorCode(),
+            ResponseCode.invalidCourseCreatorId.getErrorMessage(),
+            ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+      }
+    }
+  }
+
+  private void removeParticipants(
+      Map<String, Boolean> dbParticipants, String batchId, String courseId) {
+    CourseEnrollmentClientImpl courseEnrollmentClient = CourseEnrollmentClientImpl.getInstance();
+    dbParticipants.forEach(
+        (s, aBoolean) -> {
+          Map<String, Object> map = new HashMap<>();
+          map.put(JsonKey.USER_ID, s);
+          map.put(JsonKey.BATCH_ID, batchId);
+          map.put(JsonKey.COURSE_ID, courseId);
+          courseEnrollmentClient.createCourseUnEnrollment(
+              getActorRef(ActorOperations.UNENROLL_COURSE.getValue()), map);
+        });
+  }
+
+  private void validateCourseBatchData(CourseBatch courseBatchObject) {
+    if (ProjectUtil.isNull(courseBatchObject.getCreatedFor())
+        || courseBatchObject.getCreatedFor().isEmpty()) {
+      // throw exception since batch does not belong to any createdfor in DB ...
+      throw new ProjectCommonException(
+          ResponseCode.courseCreatedForIsNull.getErrorCode(),
+          ResponseCode.courseCreatedForIsNull.getErrorMessage(),
+          ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+    }
+    if (ProjectUtil.isNull(courseBatchObject.getEnrollmentType())
+        || !(courseBatchObject.getEnrollmentType()).equalsIgnoreCase(JsonKey.INVITE_ONLY)) {
+      throw new ProjectCommonException(
+          ResponseCode.enrollmentTypeValidation.getErrorCode(),
+          ResponseCode.enrollmentTypeValidation.getErrorMessage(),
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+  }
+
+  private List<String> getUpdatedCreatedFor(
+      List<String> createdFor, String enrolmentType, List<String> dbValueCreatedFor) {
+    if (createdFor != null) {
+      if (ProjectUtil.EnrolmentType.inviteOnly.getVal().equalsIgnoreCase(enrolmentType)) {
+        for (String orgId : createdFor) {
+          if (!dbValueCreatedFor.contains(orgId)) {
+            if (!isOrgValid(orgId)) {
+              throw new ProjectCommonException(
+                  ResponseCode.invalidOrgId.getErrorCode(),
+                  ResponseCode.invalidOrgId.getErrorMessage(),
+                  ResponseCode.CLIENT_ERROR.getResponseCode());
+            } else {
+              dbValueCreatedFor.add(orgId);
+            }
+          }
+        }
+      } else {
+        for (String orgId : createdFor) {
+          if (!dbValueCreatedFor.contains(orgId)) {
+            dbValueCreatedFor.add(orgId);
+          }
+        }
+      }
+    }
+    return dbValueCreatedFor;
+  }
+
+  private CourseBatch updateCourseBatchDate(CourseBatch courseBatch, Map<String, Object> req) {
```

Comment:
No need to return CourseBatch as in Java the object reference is passed by copy and the updates are retained after the function.